ecosystem/indexer/src/default_processor.rs

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    database::{execute_with_better_error, PgDbPool},
+    indexer::{
+        errors::TransactionProcessingError, processing_result::ProcessingResult,
+        transaction_processor::TransactionProcessor,
+    },
+    models::transactions::TransactionModel,
+    schema,
+};
+
+use aptos_rest_client::Transaction;
+use async_trait::async_trait;
+use diesel::Connection;
+use futures::future::Either;
+use std::{fmt::Debug, sync::Arc};
+
+pub struct DefaultTransactionProcessor {
+    connection_pool: PgDbPool,
+}
+
+impl DefaultTransactionProcessor {
+    pub fn new(connection_pool: PgDbPool) -> Self {
+        Self { connection_pool }
+    }
+}
+
+impl Debug for DefaultTransactionProcessor {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let state = &self.connection_pool.state();
+        write!(
+            f,
+            "DefaultTransactionProcessor {{ connections: {:?}  idle_connections: {:?} }}",
+            state.connections, state.idle_connections
+        )
+    }
+}
+
+#[async_trait]
+impl TransactionProcessor for DefaultTransactionProcessor {
+    fn name(&self) -> &'static str {
+        "default_processor"
+    }
+
+    async fn process_transaction(
+        &self,
+        transaction: Arc<Transaction>,
+    ) -> Result<ProcessingResult, TransactionProcessingError> {
+        let version = transaction.version().unwrap_or(0);
+
+        let (transaction_model, maybe_details_model, maybe_events) =
+            TransactionModel::from_transaction(&transaction);
+
+        let conn = self.connection_pool.get().map_err(|e| {
```

Comment:
nit: why not make this a helper? e.g., `let conn = self.get_connection()?`

Reply:
yeah good question- I have the `get_conn()` helper already... this may be leftover from a refactor, good catch!